src/data/init-moves.ts

```diff
@@ -0,0 +1,4085 @@
+        if (!turnCommand || !turnCommand.move) {
+          return false;
+        }
+        return (
+          turnCommand.command === BattleCommand.FIGHT
+          && !target.turnData.acted
+          && allMoves[turnCommand.move.moveId].category !== MoveCategory.STATUS
+        );
+      },
+    ),
+    new AttackMove(MoveId.MIGHTY_CLEAVE, ElementalType.ROCK, MoveCategory.PHYSICAL, 95, 100, 5, -1, 0, 9)
+      .slicingMove()
+      .ignoresProtect(),
+    new AttackMove(MoveId.TACHYON_CUTTER, ElementalType.STEEL, MoveCategory.SPECIAL, 50, -1, 10, -1, 0, 9)
+      .attr(MultiHitAttr, MultiHitType._2)
+      .slicingMove(),
+    new AttackMove(MoveId.HARD_PRESS, ElementalType.STEEL, MoveCategory.PHYSICAL, -1, 100, 10, -1, 0, 9).attr(
+      OpponentHighHpPowerAttr,
+      100,
+    ),
+    new StatusMove(MoveId.DRAGON_CHEER, ElementalType.DRAGON, -1, 15, -1, 0, 9)
+      .attr(AddBattlerTagAttr, BattlerTagType.DRAGON_CHEER, false, { failOnOverlap: true })
+      .target(MoveTarget.NEAR_ALLY),
+    new AttackMove(MoveId.ALLURING_VOICE, ElementalType.FAIRY, MoveCategory.SPECIAL, 80, 100, 10, 100, 0, 9)
+      .attr(AddBattlerTagIfBoostedAttr, BattlerTagType.CONFUSED)
+      .soundMove(),
+    new AttackMove(MoveId.TEMPER_FLARE, ElementalType.FIRE, MoveCategory.PHYSICAL, 75, 100, 10, -1, 0, 9).attr(
+      MovePowerMultiplierAttr,
+      (user, _target, _move) =>
+        user.getLastXMoves(2)[1]?.result === MoveResult.MISS || user.getLastXMoves(2)[1]?.result === MoveResult.FAIL
+          ? 2
+          : 1,
+    ),
+    new AttackMove(MoveId.SUPERCELL_SLAM, ElementalType.ELECTRIC, MoveCategory.PHYSICAL, 100, 95, 15, -1, 0, 9)
+      .attr(MissEffectAttr, crashDamageFunc)
+      .attr(NoEffectAttr, crashDamageFunc)
+      .recklessMove(),
+    new AttackMove(MoveId.PSYCHIC_NOISE, ElementalType.PSYCHIC, MoveCategory.SPECIAL, 75, 100, 10, 100, 0, 9)
+      .soundMove()
+      .attr(AddBattlerTagAttr, BattlerTagType.HEAL_BLOCK, false, { turnCountMin: 2 }),
+    new AttackMove(MoveId.UPPER_HAND, ElementalType.FIGHTING, MoveCategory.PHYSICAL, 65, 100, 15, 100, 3, 9)
+      .attr(FlinchAttr)
+      .condition(new UpperHandCondition()),
+    new AttackMove(MoveId.MALIGNANT_CHAIN, ElementalType.POISON, MoveCategory.SPECIAL, 100, 100, 5, 50, 0, 9).attr(
+      StatusEffectAttr,
+      StatusEffect.TOXIC,
+    ),
+  ];
+
+  for (const move of rawAllMoves) {
+    // Make sure `allMoves` assigns correct ID to every move
+    allMoves[move.id] = move;
+    addFireMovesThawFrozenTargetAttribute(move);
+  }
+}
+
+//#region Helpers
+
+/**
+ * All damaging Fire-type moves can now thaw a frozen target, regardless of whether or not they have a chance to burn.
+ * @source {@link https://bulbapedia.bulbagarden.net/wiki/Freeze_(status_condition) | Bulbapedia - Freeze (Status Condition)}
+ */
+function addFireMovesThawFrozenTargetAttribute(move: Move) {
+  if (move.type === ElementalType.FIRE) {
```

Comment:
```suggestion
  if (move.type === ElementalType.FIRE && move.isAttackMove()) {
```
I just realized that this logic was originally in the constructor for `AttackMove`, not `Move`